cts-java/src/test/java/org/ga4gh/cts/api/endpoints/RawEndpointsIT.java

```diff
@@ -0,0 +1,123 @@
+package org.ga4gh.cts.api.endpoints;
+
+import com.mashape.unirest.http.Unirest;
+import com.mashape.unirest.http.exceptions.UnirestException;
+import junitparams.JUnitParamsRunner;
+import junitparams.Parameters;
+import org.ga4gh.ctk.transport.TransportUtils;
+import org.ga4gh.ctk.transport.URLMAPPING;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import java.net.HttpURLConnection;
+
+import static org.assertj.core.api.StrictAssertions.assertThat;
+
+/**
+ * Tests for the properties of the raw server endpoints.
+ *
+ * @author Herb Jellinek
+ */
+@RunWith(JUnitParamsRunner.class)
+@Category(EndpointsTests.class)
+public class RawEndpointsIT {
+
+    private static final URLMAPPING urls = URLMAPPING.getInstance();
+
+    /**
+     * Given a partial URL, use {@link TransportUtils#makeUrl(String, String)} to combine it with
+     * the root URL and return the result.
+     * @param partialUrl a partial URL
+     * @return a full URL including the root URL
+     */
+    private static String makeFullUrl(String partialUrl) {
+        return TransportUtils.makeUrl(urls.getUrlRoot(), partialUrl);
+    }
+
+    /**
+     * Test that the basic server verbs/methods work as expected.
+     *
+     * @param fullUrl the URL to test
+     */
+    private void testHttpMethods(String fullUrl) throws UnirestException {
+        assertThat(Unirest.get(fullUrl).asBinary().getStatus()).isEqualTo(HttpURLConnection.HTTP_BAD_METHOD);
+        assertThat(Unirest.options(fullUrl).asBinary().getStatus()).isEqualTo(HttpURLConnection.HTTP_OK);
+        assertThat(Unirest.post(fullUrl).asBinary().getStatus()).isEqualTo(HttpURLConnection.HTTP_UNSUPPORTED_TYPE);
+    }
+
+    /**
+     * Test the status codes we're supposed to receive from the GET, POST, and OPTIONS methods on
+     * <tt>/callsets/search</tt>.
+     *
+     * <p>See server/tests/unit/test_views.py: testRouteCallsets</p>
```

Comment:
I think if it's not currently documented (and not obviously the right thing to do) we should remove the test. Opening an issue on it for the schemas so that it is documented, and incorporating the test afterwards is the way to go I think. The baseline should be uncontroversial stuff, and we don't want undocumented assumptions made about the HTTP behaviour becoming requirements.